### PR TITLE
Updates to VGG, DataLoader and metric calculator

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #(issue_number)

## Proposed Changes

- VGG no longer depends on batch size
- Data loader is handling elastic and padding better
- Validation metrics (in utils) are getting generated better for regression/classification tasks that have a label

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GANDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`

### CODE REVIEW

File: GANDLF/losses.py [hunk 1]
```diff
@@ -61,7 +65,12 @@ def MCD(pm, gt, num_class, weights = None, ignore_class = None):
         
         if calculate_dice_for_label:
             currentDice = dice(gt[:, i, ...], pm[:, i, ...])
-            # currentDiceLoss = 1 - currentDice # subtract from 1 because this is a loss
+            if loss_type == 1:
+                currentDice = 1 - currentDice # subtract from 1 because this is a loss
+            elif loss_type == 2:
+                currentDice = -torch.log(
+                    currentDice + torch.finfo(torch.float32).eps
+                )  # negative because we want positive losses
```
Comment (Geeks-Sid) [inline GANDLF/losses.py:73]: Not a fan of this. maybe we can provide log_dice? This makes things unclear.

File: GANDLF/parseConfig.py [hunk 2]
```diff
@@ -131,6 +131,37 @@ def parseConfig(config_file_path, version_check = True):
...
+      else:
+        initialize_threshold = True
+      
+      if initialize_threshold:
+        temp_dict['accuracy']['threshold'] = 0.5
+    params['metrics'] = temp_dict
+      
+  else:
+    sys.exit('The key \'metrics\' needs to be defined')
+
```
Comment (Geeks-Sid) [inline GANDLF/parseConfig.py:164]: This needs to be standardised to 4 spaces.

File: GANDLF/utils.py [hunk 3]
```diff
@@ -273,13 +353,30 @@ def get_metrics_save_mask(model, device, loader, psize, channel_keys, value_keys
...
+        # icc_3_1 = icc.icc(icc_input, model='twoway', type='consistency', unit='single')
+        # icc_3_k = icc.icc(icc_input, model='twoway', type='consistency', unit='average')
+        # print('=== ICC ===')
+        # print('icc_1_1:', icc_1_1)
+        # print('icc_1_k:', icc_1_k)
+        # print('icc_2_1:', icc_2_1)
+        # print('icc_2_k:', icc_2_k)
+        # print('icc_3_1:', icc_3_1)
+        # print('icc_3_k:', icc_3_k)
+        # print('=== ICC ===')
```
Comment (Geeks-Sid) [inline GANDLF/utils.py:379]: Needs to be removed before merge.

File: gandlf_padder [hunk 4]
```diff
@@ -0,0 +1,172 @@
+#!usr/bin/env python
```
Comment (Geeks-Sid) [inline gandlf_padder:1]: Not a fan, this will be temporary as we discussed.

**Review by Geeks-Sid — APPROVED** [review]

File: GANDLF/losses.py (hunk 1 above)
Comment (sarthakpati) [reply GANDLF/losses.py:73]: I just didn't want to have 2 implementations of dice loss floating around. This is the base function, in any case, so I figured it would be okay. What do you think?

File: GANDLF/utils.py (hunk 3 above)
Comment (sarthakpati) [reply GANDLF/utils.py]: Done.

File: GANDLF/parseConfig.py (hunk 2 above)
Comment (sarthakpati) [reply GANDLF/parseConfig.py]: Blacked this file for consistency.

**Review by Geeks-Sid — APPROVED** [review]
